homeassistant/components/light/mqtt_template.py

```diff
@@ -0,0 +1,252 @@
+            elif state == STATE_OFF:
+                self._state = False
+            else:
+                _LOGGER.warning('Invalid state value received')
+
+            # read brightness
+            if self._brightness is not None:
+                try:
+                    self._brightness = int(
+                        self._templates[CONF_BRIGHTNESS_TEMPLATE].
+                        render_with_possible_json_value(payload)
+                    )
+                except ValueError:
+                    _LOGGER.warning('Invalid brightness value received')
+
+            # read color
+            if self._rgb is not None:
+                try:
+                    self._rgb[0] = int(
+                        self._templates[CONF_RED_TEMPLATE].
+                        render_with_possible_json_value(payload))
+                    self._rgb[1] = int(
+                        self._templates[CONF_GREEN_TEMPLATE].
+                        render_with_possible_json_value(payload))
+                    self._rgb[2] = int(
+                        self._templates[CONF_BLUE_TEMPLATE].
+                        render_with_possible_json_value(payload))
+                except ValueError:
+                    _LOGGER.warning('Invalid color value received')
+
+            self.update_ha_state()
+
+        if self._topics[CONF_STATE_TOPIC] is not None:
+            mqtt.subscribe(self._hass, self._topics[CONF_STATE_TOPIC],
+                           state_received, self._qos)
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return self._brightness
+
+    @property
+    def rgb_color(self):
+        """Return the RGB color value [int, int, int]."""
+        return self._rgb
+
+    @property
+    def should_poll(self):
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return True if entity is on."""
+        return self._state
+
+    @property
+    def assumed_state(self):
+        """Return True if unable to access real state of the entity."""
+        return self._optimistic
+
+    def turn_on(self, **kwargs):
+        """Turn the entity on."""
+        # state
+        values = {'state': True}
+        if self._optimistic:
+            self._state = True
+
+        # brightness
+        if ATTR_BRIGHTNESS in kwargs:
+            values['brightness'] = int(kwargs[ATTR_BRIGHTNESS])
+
+            if self._optimistic:
+                self._brightness = kwargs[ATTR_BRIGHTNESS]
+
+        # color
+        if ATTR_RGB_COLOR in kwargs:
+            values['red'] = kwargs[ATTR_RGB_COLOR][0]
+            values['green'] = kwargs[ATTR_RGB_COLOR][1]
+            values['blue'] = kwargs[ATTR_RGB_COLOR][2]
```

Comment:
`values['color'] = kwargs[ATTR_RGB_COLOR]` is also possible here. Let me know what you prefer.